### DISCUSSION TITLE
Rules with a body property containing a dot in its name

### DISCUSSION THREAD

**Discussion by 14geronimo** [mockoon/mockoon#739] [Help] [answered]
I need to create a Rule on a value `myValue` for the XML attribute at this Xpath: `/server/requests/CustomServer.processRq/@xslt`

If the third nested element didn't have a dot in its name, e.g. `CustomServerProcessRq` instead of `CustomServer.processRq`, then this Rule works perfectly:
```
Body  |  server.requests.CustomServerProcessRq._attributes.xslt  |  equals  |  myValue
```

However I can't figure out a way to express this property in object-path because of the dot in the element name. I've tried these in Mockoon, to no avail:
- `server.requests.CustomServer.ProcessRq._attributes.xslt`
- `["server", "requests", "CustomServer.ProcessRq", "_attributes", "xslt"]`
- `server.requests.[CustomServer.ProcessRq]._attributes.xslt`
- `server.requests."CustomServer.ProcessRq"._attributes.xslt`
- `server.requests.'CustomServer.ProcessRq'._attributes.xslt`
- `server.requests.CustomServer\.ProcessRq._attributes.xslt`
- `server.requests.CustomServer\u002EProcessRq._attributes.xslt`  (U+002E is unicode for the dot character)
- `server.requests.CustomServer\u002E ProcessRq._attributes.xslt`
- `server.requests.\u0043\u0075\u0073\u0074\u006F\u006D\u0053\u0065\u0072\u0076\u0065\u0072\u002E\u0070\u0072\u006F\u0063\u0065\u0073\u0073\u0052\u0071._attributes.xslt`  (the unicode representation of string "CustomServer.processRq")
- `server.requests.\u0043 \u0075 \u0073 \u0074 \u006F \u006D \u0053 \u0065 \u0072 \u0076 \u0065 \u0072 \u002E \u0070 \u0072 \u006F \u0063 \u0065 \u0073 \u0073 \u0052 \u0071 ._attributes.xslt`
- `server.requests.\u0043 \u0075 \u0073 \u0074 \u006F \u006D \u0053 \u0065 \u0072 \u0076 \u0065 \u0072 \u002E \u0070 \u0072 \u006F \u0063 \u0065 \u0073 \u0073 \u0052 \u0071._attributes.xslt`

Are dotted element names supported at all by Mockoon, or would this be a request for improvement?

**Comment by 255kb** [comment]
Hi @14geronimo 
This has been requested and discussed before (#338). 
It is currently not supported (due to the lib we are using to retrieve data in objects), and there is currently no plan to make changes to support dots in body properties.

**Reply by 14geronimo** [reply]
I appreciate the pointer to the previous posts on this question @255kb, thank you. My query relates to body element names, where dot also indicate parent elements, but I guess the issue is indeed similar to dots in query param names.

I would still suggest to revisit the decision to not support this. Dotted names are not uncommon at all. Besides, there is no need to introduce custom syntax either, object-path supports two mechanisms to handle this:
- the array notation: e.g. `["a", "dot.dot"]` as listed in the object-path README
- the use of unicode: e.g. `a.\u1200` also listed in the object-path README
ref. [https://github.com/mariocasciaro/object-path](https://github.com/mariocasciaro/object-path), Usage section, "get deep property" examples

Mockoon appears to handle the `a.b` syntax correctly, but neither of the two alternative methods available with open-path. With thrid-party APIs, you don't get to choose query params and body element names unfortunately, and so many of them do use dotted names (not to mention other weirdnesses that would benefit from the alternative open-path syntaxes) that adding support for these in Mockoon would greatly widen the scope where the (otherwise brilliant) Rule feature can be used.

**Reply by 255kb** [reply, marked as answer]
Ok, you convinced me, let's reopen it. We will have to investigate a bit and find a way to support the syntax as it is a text input. 
Maybe detect if brackets are present and convert to an array of strings. 
I just don't see why using `\u002E` is not working. It will be part of the investigation I guess :)
